pkg/social/common.go

```diff
@@ -18,3 +22,25 @@ func isEmailAllowed(email string, allowedDomains []string) bool {
 
 	return valid
 }
+
+func HttpGet(client *http.Client, url string) ([]byte, error) {
+	r, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	defer r.Body.Close()
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode >= 300 {
+		return nil, fmt.Errorf(string(body))
+	}
+
+	log.Trace("HTTP GET %s: %s %s", url, r.Status, string(body))
```

Comment:
using the log package static functions is something I am trying to move away from (should be removed entirely as I think there are almost no users of them anymore), instead instantiate a package level logger like `var oauthLogger log.Logger =  log.New("oauth")`, that logger interface supports the log15 key/value based logging interface